### ISSUE TITLE
[Feature request] Handlers for file types

### ISSUE THREAD

**Issue by JJK96** [fanglingsu/vimb#474]
Open specific file types (for example, pdf) in an external program instead of downloading them and then having to manually open them in that program.

**Comment by cdlscpmv** [comment]
In the WebKit1 version of vimb there was an ability to call an external program to handle all downloads. So you could write a wrapper which, depending of the filetype of a downloaded file, could either open it in an external program (a pdf viewer), or simply download it with wget. You could also download the file first, and then automatically open it with an external program (if your pdf viewer can't hande loading it from the Internet).

**Comment by fanglingsu** [comment]
Yes there was an external downloaded in webkit1 version, which we want to port also to the webkit2 version #348. In case the download is done by vimb we could use the `DownloadFinished` event of the autocommands to trigger some script. But this would require to provide some information about the downloaded file mime-type, download location (possibly by environment variable for `:shell` command but this might not work properly with multiple concurrent downloads).

**Comment by Unixsys** [comment]
Perhaps a config option could be added that specifies a script to run the file through? And provide an example script, too.

**Comment by Moempfred** [comment]
It's not perfect but in most cases it works properly.
Autocommand (in ~/.config/vimb/config)
```
aug open_download
    au DownloadFinished * sh! ~/dwnld-opener
aug end
```
Opener script
```
cd ~/Downloads 
xdg-open "$(ls -Art ~/Downloads | tail -n 1)"
cd -
```
